src/gui/DatabaseOpenWidget.cpp

```diff
@@ -168,6 +167,10 @@ void DatabaseOpenWidget::enterKey(const QString& pw, const QString& keyFile)
 void DatabaseOpenWidget::openDatabase()
 {
     KeePass2Reader reader;
+    if ((m_ui->editPassword->text().isEmpty()) && (!m_ui->checkKeyFile->isChecked())) {
```

Comment:
Does it make much sense to check if the key file is checked?

Reply:
What about removing this check and just call `activatePassword()`?

This code should be called only when a db file is selected for opening. (if I'm not wrong)
An the default is empty password and no keyfile.

If the `Remember last keyfile and security dongle` option is enabled keyfile will be autofilled but enforcing a password it's not bad afterall (I hope nobody uses a db with no password or keyfile-only)

What do you think?

Reply:
Well, dude, you are right! Why not just check the Password checkbox by default? I will update this PR following that aim.